packages/gatsby/src/utils/webpack/get-ssr-chunk-hashes.ts

```diff
@@ -17,16 +17,16 @@ function getHashes(
     hashes.push(chunk.hash)
   }
 
-  for (const childChunkGroups of Object.values(
-    chunkGroup.getChildrenByOrders(
-      compilation.moduleGraph,
-      compilation.chunkGraph
+  for (const childChunkGroup of chunkGroup.childrenIterable) {
+    const isNotImportedByAsyncRequires = childChunkGroup.origins.every(
+      origin => !origin.module.identifier().includes(`async-requires`)
     )
-  )) {
-    for (const childChunkGroup of childChunkGroups) {
+
+    if (isNotImportedByAsyncRequires) {
       getHashes(childChunkGroup, compilation, hashes)
     }
   }
+
```

Comment:
`chunkGroup.getChildrenByOrders` does iterate on `childrenIterable` but it also does some filtering which excludes those dynamic imports for some reason.

So we are using `childrenIterable` directly here - it is sorted set (by `id`, so it might not be very stable?) and just ignore dynamic imports created in `async-requires`, otherwise any change anywhere would change "shared" hash resulting in invalidating all html